mailpoet/tests/integration/_bootstrap.php

```diff
@@ -30,50 +30,16 @@
 $console->writeln('Loading WP core... (' . $wpLoadFile . ')');
 
 $console->writeln('Cleaning up database...');
-$models = [
-  'CustomField',
-  'Newsletter',
-  'NewsletterSegment',
-  'NewsletterOption',
-  'NewsletterOptionField',
-  'Segment',
-  'ScheduledTask',
-  'ScheduledTaskSubscriber',
-  'SendingQueue',
-  'Subscriber',
-  'SubscriberCustomField',
-  'SubscriberSegment',
-  'StatisticsOpens',
-  'StatisticsClicks',
-  'StatisticsNewsletters',
-];
-
-$entities = [
-  MailPoet\Entities\FormEntity::class,
-  MailPoet\Entities\NewsletterLinkEntity::class,
-  MailPoet\Entities\NewsletterTemplateEntity::class,
-  MailPoet\Entities\SettingEntity::class,
-  MailPoet\Entities\StatisticsUnsubscribeEntity::class,
-];
 
 $connection = ContainerWrapper::getInstance(WP_DEBUG)->get(Connection::class);
-$destroy = function($model) use ($connection) {
-  $modelName = '\MailPoet\Models\\' . $model;
-  if (!class_exists($modelName)) {
-    throw new \RuntimeException("Class $modelName doesn't exist.");
-  }
-  $class = new \ReflectionClass($modelName);
-  $table = $class->getStaticPropertyValue('_table');
-  $connection->executeUpdate("TRUNCATE $table");
-};
-array_map($destroy, $models);
-
 $entityManager = ContainerWrapper::getInstance(WP_DEBUG)->get(EntityManager::class);
-foreach ($entities as $entity) {
-  $tableName = $entityManager->getClassMetadata($entity)->getTableName();
-  $connection->query('SET FOREIGN_KEY_CHECKS=0');
-  $connection->executeUpdate("TRUNCATE $tableName");
-  $connection->query('SET FOREIGN_KEY_CHECKS=1');
+$entitiesMeta = $entityManager->getMetadataFactory()->getAllMetadata();
+foreach ($entitiesMeta as $entityMeta) {
```

Comment:
Would this fetch the old Paris ORM models?

If not, Can we keep the function truncating the tables around till all the models have been migrated?

Reply:
Thanks for the review, @triple0t! This doesn't fetch old Paris ORM models, but unless I'm missing something it is not a problem. My understanding is that what we want to do in this part of the code is truncate all the MailPoet tables. So it doesn't matter if we are getting all the tables from Paris or Doctrine as long as we get all the tables. As you pointed out above, some of the old Paris models are still being used indeed, but we already have correspondent Doctrine entities for those models.

As pointed out in the commit message, I opted to include this change here to make it easier when removing future models and to make sure the list of tables is always updated when we add new tables in the future.

What do you think?

Reply:
Understood.
Thanks for the explanation, Rodrigo.